File: bulk/csharp/RaceCondition01.cs
// CWE-362: Race Condition on shared balance
public class RaceCondition01 {
    private int _balance = 1000;
    public void Withdraw(int amount) {
        if (_balance >= amount) { // check
            Thread.Sleep(1); // simulate work
            _balance -= amount; // deduct — not synchronized
        }
    }
}

File: bulk/java/RaceCondition01.java
// CWE-362: Race Condition in account balance check
public class RaceCondition01 {
    private int balance = 500;
    public void withdraw(int amount) {
        if (balance >= amount) { // check — not synchronized
            try { Thread.sleep(10); } catch (InterruptedException e) {}
            balance -= amount; // deduct — race allows overdraft
        }
    }
}
